File: built-in plugins dsp/mb-deesser.cpp
/**
 * MB De-Esser
 * Category : effect
 * Type     : compressor
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Sibilance reduction
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_DEESSER_H
#define MB_DEESSER_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbDeesser : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-deesser";
    static constexpr const char* PLUGIN_NAME    = "MB De-Esser";
    static constexpr const char* PLUGIN_TYPE    = "compressor";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float freq = 6000f;  // range [3000, 12000]
    float threshold = -20f;  // range [-60, 0]
    float reduction = 6f;  // range [0, 24]
    };

    MbDeesser() = default;
    ~MbDeesser() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.freq = std::clamp(params.freq, 3000f, 12000f);
        params.threshold = std::clamp(params.threshold, -60f, 0f);
        params.reduction = std::clamp(params.reduction, 0f, 24f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB De-Esser
        return input;
    }
};

#endif // MB_DEESSER_H

File: built-in plugins dsp/mb-deesser.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbDeesserPlugin: PluginDefinition = { id: 'mb-deesser', slug: 'mb-deesser', name: 'MB De-Esser', category: 'effect', type: 'compressor', version: '1.0.0', description: 'Sibilance reduction', author: 'Max Booster', parameters: [{ id: 'freq', name: 'Frequency', type: 'float', defaultValue: 6000, minValue: 3000, maxValue: 12000, automatable: true }, { id: 'threshold', name: 'Threshold', type: 'float', defaultValue: -20, minValue: -60, maxValue: 0, automatable: true }, { id: 'reduction', name: 'Reduction', type: 'float', defaultValue: 6, minValue: 0, maxValue: 24, automatable: true }], defaultPreset: { freq: 6000, threshold: -20, reduction: 6 } };

export default MbDeesserPlugin;
